styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodPollerTest.java

```diff
@@ -0,0 +1,153 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.docker;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.spotify.styx.model.Event;
+import com.spotify.styx.model.WorkflowInstance;
+import com.spotify.styx.monitoring.Stats;
+import com.spotify.styx.state.RunState;
+import com.spotify.styx.state.StateData;
+import com.spotify.styx.state.StateManager;
+import com.spotify.styx.testdata.TestData;
+import io.fabric8.kubernetes.api.model.DoneablePod;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.ClientMixedOperation;
+import io.fabric8.kubernetes.client.dsl.ClientPodResource;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KubernetesDockerRunnerPodPollerTest {
+
+  private static final String POD_NAME = "test-pod-1";
+  private static final String POD_NAME_2 = "test-pod-2";
+  private static final WorkflowInstance WORKFLOW_INSTANCE =
+      WorkflowInstance.create(TestData.WORKFLOW_ID, "foo");
+  private static final WorkflowInstance WORKFLOW_INSTANCE_2 =
+      WorkflowInstance.create(TestData.WORKFLOW_ID_2, "bar");
+  private static final DockerRunner.RunSpec RUN_SPEC =
+      DockerRunner.RunSpec.create("busybox", ImmutableList.of(), Optional.empty());
+
+  @Mock
+  KubernetesClient k8sClient;
+  @Mock
+  ClientMixedOperation<Pod, PodList, DoneablePod, ClientPodResource<Pod, DoneablePod>> pods;
+  PodList podList;
+
+  @Mock
+  StateManager stateManager;
+  @Mock
+  Stats stats;
+
+  KubernetesDockerRunner kdr;
+
+  @Before
+  public void setUp() throws Exception {
+    when(k8sClient.inNamespace(any(String.class))).thenReturn(k8sClient);
+    when(k8sClient.pods()).thenReturn(pods);
+
+    kdr = new KubernetesDockerRunner(k8sClient, stateManager, stats);
+    podList = new PodList();
+  }
+
+  @Test
+  public void shouldSendRunErrorWhenPodForRunningWFIDoesntExist() throws Exception {
+    Pod createdPod = KubernetesDockerRunner.createPod(WORKFLOW_INSTANCE, RUN_SPEC);
+    podList.setItems(Arrays.asList(createdPod));
+    when(k8sClient.pods().list()).thenReturn(podList);
+    setupActiveInstances(RunState.State.RUNNING);
+
+    kdr.pollPods();
+
+    verify(stateManager, times(1)).receiveIgnoreClosed(
```

Comment:
I wouldn't like to reverify for something that there is already a unit test `shouldNotSendRunErrorWhenPodForRunningWFIExists`. I don't think it adds value to improving test coverage.